src/components/ModalDialog/ModalDialog.vue

```diff
@@ -0,0 +1,282 @@
+<template>
+  <section
+    v-if="modelValue"
+    class="unnnic-modal-dialog"
+  >
+    <section
+      class="unnnic-modal-dialog__overlay"
+      @click.stop="!persistent && close()"
+    />
+    <section
+      :class="[
+        'unnnic-modal-dialog__container',
+        `unnnic-modal-dialog__container--${size}`,
+      ]"
+    >
+      <header
+        v-if="title"
+        class="unnnic-modal-dialog__container__header"
+      >
+        <section class="unnnic-modal-dialog__container__title-container">
+          <UnnnicIcon
+            v-if="icon || type"
+            class="unnnic-modal-dialog__container__title-icon"
+            :icon="icon || iconsMapper[type]?.icon"
+            :scheme="iconScheme || iconsMapper[type]?.scheme"
+            size="md"
+          />
+          <h1 class="unnnic-modal-dialog__container__title-text">
+            {{ title }}
+          </h1>
+        </section>
+        <UnnnicIcon
+          v-if="showCloseIcon"
+          icon="close"
+          clickable
+          @click="close()"
+        />
+      </header>
+      <section class="unnnic-modal-dialog__container__content">
+        <slot></slot>
+      </section>
+      <section
+        v-if="primaryButtonText"
+        :class="[
+          'unnnic-modal-dialog__container__actions',
+          showActionsDivider &&
+            'unnnic-modal-dialog__container__actions--divider',
+        ]"
```

Comment:
The way it was implemented may be less clear and more difficult to scale than using an object to define the conditional classes. I recommend that you use one of the options below:

```
        :class="{
          'unnnic-modal-dialog__container__actions': true,
          'unnnic-modal-dialog__container__actions--divider': showActionsDivider
        }"
```

or


```
        :class="[
          'unnnic-modal-dialog__container__actions',
          {'unnnic-modal-dialog__container__actions--divider': showActionsDivider}
        ]"
```